lecture/src/main/java/de/tum/in/www1/artemis/lecture/web/rest/LectureResource.java

```
@@ -0,0 +1,251 @@
+            if (lectureUnit == null) {
+                return false;
+            }
+            if (lectureUnit instanceof ExerciseUnit) {
+                return ((ExerciseUnit) lectureUnit).getExercise() != null && authCheckService.isAllowedToSeeLectureUnit(lectureUnit, user)
+                        && exercisesWithAllInformationNeeded.contains(((ExerciseUnit) lectureUnit).getExercise());
+            }
+            else if (lectureUnit instanceof AttachmentUnit) {
+                return ((AttachmentUnit) lectureUnit).getAttachment() != null && authCheckService.isAllowedToSeeLectureUnit(lectureUnit, user);
+            }
+            else {
+                return authCheckService.isAllowedToSeeLectureUnit(lectureUnit, user);
+            }
+        }).peek(lectureUnit -> {
+            if (lectureUnit instanceof ExerciseUnit) {
+                Exercise exercise = ((ExerciseUnit) lectureUnit).getExercise();
+                // we replace the exercise with one that contains all the information needed for correct display
+                exercisesWithAllInformationNeeded.stream().filter(exercise::equals).findAny().ifPresent(((ExerciseUnit) lectureUnit)::setExercise);
+            }
+        }).collect(Collectors.toList());
+
+        filteredLecture.setLectureUnits(lectureUnitsUserIsAllowedToSee);
+        return filteredLecture;
+    }
+
+    /**
+     * DELETE /lectures/:id : delete the "id" lecture.
+     *
+     * @param lectureId the id of the lecture to delete
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @DeleteMapping("/lectures/{lectureId}")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<Void> deleteLecture(@PathVariable Long lectureId) {
+        Optional<Lecture> optionalLecture = lectureRepository.findByIdWithPostsAndLectureUnitsAndLearningGoals(lectureId);
+        if (optionalLecture.isEmpty()) {
+            return ResponseEntity.notFound().build();
```

Could you throw a detailed exception here?